src/goto-instrument/dump_c.cpp

```diff
@@ -1062,6 +1224,10 @@ void dump_ct::convert_function_declaration(
   if(symbol.name!=goto_functionst::entry_point() &&
      symbol.name!=ID_main)
   {
+    // make sure typedef names are available before outputting this
+    // declaration
+    collect_typedefs(symbol.type, true);
```

Comment:
> Also I think "2" is addressed by an additional comment that I've inserted above that line

Well maybe, for me I read it as a justification to call `colllect_typedefs`, rather than to exclude collecting typedefs when looking at main. It also seems inconsistent, if user runs with `--function foo` we _will_ collect typedefs from `foo` but not from `main`. 

If you think this is correct, could you add a `typedef` within the main function?